### PR TITLE
Feature/api tests

### PR DESCRIPTION
**Pull request by armins**
Add some API tests to try to reproduce an empty result we saw today when debugging manually with docker.

### CODE REVIEW

**Review by llinauer — DISMISSED** [review]
Looks good. Maybe try monkeypatch as suggested.

File: natural_language_processing/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+def test_empty_input(client):
+    response = client.post("/", json={"text": ""})
+    assert response.status_code == 400
+    data = response.get_json()
+    assert isinstance(data, dict)
+    assert data["error"] == "No text provided for NER extraction"
+
+def test_invalid_json(client):
+    response = client.post("/", data="notjson", content_type="application/json")
+    assert response.status_code in (400, 422)  # Adjust for your error handling
```
[the remark this answers is not in the document]
Comment (b3n4kh) [reply natural_language_processing/tests/test_api.py:39]: I think it should error! Input is not supposed to be empty!

[the remark this answers is not in the document]
Comment (llinauer) [reply natural_language_processing/tests/test_api.py:39]: Yes, I think expecting 400 is fine.

However, when I run this test "in the field" with 
`curl -X POST http://127.0.0.1:8000/  -H "Content-Type: application/json"   -d '"notjson"'`

I get a 500 with the traceback:

"natural_language_processing/router.py", line 18, in post
    text = data.get("text", "")
           ^^^^^^^^
AttributeError: 'str' object has no attribute 'get'

Which is expected, sice I sent the string "notjson", which does not have a "get" method.
I am wondering why the test passes. Shouldn't it fail because it expects a 400 and gets a 500?

[the remark this answers is not in the document]
Comment (armins) [reply natural_language_processing/tests/test_api.py:39]: will check

**Review by llinauer — APPROVED** [review]
